### ISSUE TITLE
Incorrect Maven Snapshot file resolution

### ISSUE THREAD

**Issue by fommil** [sbt/sbt#679]
## steps

``` scala
lazy val root = (project in file(".")).
  settings(
    libraryDependencies += "com.github.fommil" % "java-logging" % "1.2-SNAPSHOT",
    resolvers += Resolver.sonatypeRepo("public")
  )
```
## original report

I have discussed this on the mailing list, but not received any response to date:

https://groups.google.com/d/topic/simple-build-tool/ayKrOWrBF94/discussion

Basically, when depending on this project which is published on sonatype snapshots

https://oss.sonatype.org/content/repositories/snapshots/com/github/fommil/java-logging/1.0-SNAPSHOT/maven-metadata.xml

SBT seems to incorrectly calculate the name of the jar files. Instead of simply reading them off the metadata file, SBT appears to be calculating them from the timestamp information. There is no guarantee these all agree, and indeed this is an example of where they do not.

I'm using sbt-0.12.2 on OS X.

**Comment by fommil** [comment]
Closing the loop: http://stackoverflow.com/questions/15055688

**Comment by ediweissmann** [comment]
+1

**Comment by eed3si9n** [comment]
Sorry for the lack of response from the sbt team. New management here.
First, I'd like to understand the problem. Could you rephrase this case into a reproducible steps or a test project? If the problem is Sonatype not being truthful about their snapshot build number, what do you expect sbt and/or Ivy to do?

**Comment by fommil** [comment]
the mailing list post should contain everything you need. I worked around this by creating builds that have the build numbers in a manner that sbt understands. Apologies, the sonatype snapshot has been purged.

**Comment by jsuereth** [comment]
@fommil I have a new branch which is using Aether to resolve maven artifacts. I believe this may fix your issue, but I was wondering if you still have something which can reproduce the problem so we can test it out (or create a test case).

Thanks!

**Comment by fommil** [comment]
cool!!! Hopefully Aether is faster at resolving and downloading. I changed my build pipeline and never got the pom.xml files looking quite this way ever again. Unless someone else has this, I'm happy to close and flag this as an upstream bug that has potentially been fixed by Aether.

**Comment by eed3si9n** [comment]
#2075 (read `maven-metadata.xml`) might fix this bug without sbt-maven-resolver.

```
[info] downloading https://oss.sonatype.org/content/repositories/public/com/github/fommil/java-logging/1.2-SNAPSHOT/java-logging-1.2-20140728.161141-1.jar ...
```
